### PR TITLE
Fix number of NULL ptr issues in ESYS

### PR DESCRIPTION
**Pull request by tstruk**
esys: Check for NULL auth in iesys_compute_session_value
esys: Fix NULL ptr auth in Esys_TR_SetAuth
esys: handle NULL sequenceHandleNode
test: esys: Add test for Esys_TR_SetAuth with NULL auth

### CODE REVIEW

File: src/tss2-esys/api/Esys_HMAC_Start.c [hunk 1]
```diff
@@ -25,12 +25,12 @@ static void store_input_parameters (
     esysContext->in.HMAC_Start.handle = handle;
     esysContext->in.HMAC_Start.hashAlg = hashAlg;
     if (auth == NULL) {
-        esysContext->in.HMAC_Start.auth = NULL;
```
Comment (williamcroberts) [inline src/tss2-esys/api/Esys_HMAC_Start.c]: Commit message: sed -i s/esey/esys

I wish github allowed comments on the commit message itself.

File: src/tss2-esys/esys_tr.c [hunk 2]
```diff
@@ -369,6 +369,7 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
     RSRC_NODE_T *esys_object;
     TSS2_RC r;
     _ESYS_ASSERT_NON_NULL(esys_context);
+    _ESYS_ASSERT_NON_NULL(authValue);
```
Comment (williamcroberts) [inline src/tss2-esys/esys_tr.c]: I'm not 100% sure this is the right fix. IIRC, authValue NULL implies 0 length auth buffer.

Comment (williamcroberts) [reply src/tss2-esys/esys_tr.c]: Commit message body lacking justification

Comment (tstruk) [reply src/tss2-esys/esys_tr.c]: Right, just checked the spec and if auth is NULL the auth len should be set to zero. Patch updated.

File: src/tss2-esys/esys_tr.c [hunk 3]
```diff
@@ -369,10 +369,16 @@ Esys_TR_SetAuth(ESYS_CONTEXT * esys_context, ESYS_TR esys_handle,
     RSRC_NODE_T *esys_object;
     TSS2_RC r;
     _ESYS_ASSERT_NON_NULL(esys_context);
+
```
Comment (williamcroberts) [inline src/tss2-esys/esys_tr.c]: whitespace change: Newline added.....
